1051 state local ID

tstrass opened the pull request:
For [SARAALERT-1051](https://tracker.codev.mitre.org/browse/SARAALERT-1051).

Corresponding implementation: https://github.com/SaraAlert/SaraAlert/pull/558

This does two things:
- Migrates the project to the current version of sushi per https://fshschool.org/docs/sushi/migration/
- Adds a slice for our State/Local ID identifier. I separated this into the second commit so that you can see the diff better.

tstrass committed:
Migrate project to new sushi version

Most involved moving/renaming files. Also gets the new versions
of the publisher and genonce scripts

tstrass committed:
Add identifier slice for state/local ID

ngfreiter commented on ig.ini:
```diff
@@ -0,0 +1,12 @@
+[IG]
+; ***********************************************************************************************
+; *                               WARNING: DO NOT EDIT THIS FILE                                *
+; *                                                                                             *
+; * This file is generated by SUSHI. Any edits you make to this file will be overwritten.       *
+; *                                                                                             *
+; * This ig.ini was generated using the template property in config.yaml. To provide your own   *
+; * ig.ini, create an ig.ini file in the ig-data folder with required properties: ig, template. *
+; * See: https://build.fhir.org/ig/FHIR/ig-guidance/using-templates.html#root                   *
+; ***********************************************************************************************
```
We can ditch this warning, since it is now not true.

tstrass replied:
https://github.com/SaraAlert/saraalert-fhir-ig/pull/6/commits/e90a101d4dc0c7a2e8219aacd6e9bf7cf6da4d62

ngfreiter commented on input/ImplementationGuide-saraalert-fhir.json:
```diff
@@ -0,0 +1,238 @@
+{
```
We don't have to check this file into source control, since SUSHI will generate it to `fsh-generated/resources`. So we can remove this file, and then we just have to change the `ig` line in `ig.ini` to be:
```
ig = fsh-generated/resources/ImplementationGuide-saraalert-fhir.json
```

tstrass replied:
https://github.com/SaraAlert/saraalert-fhir-ig/pull/6/commits/e90a101d4dc0c7a2e8219aacd6e9bf7cf6da4d62

ngfreiter commented on ig.ini:
```diff
@@ -0,0 +1,12 @@
+; * See: https://build.fhir.org/ig/FHIR/ig-guidance/using-templates.html#root                   *
+; ***********************************************************************************************
+ig = input/ImplementationGuide-saraalert-fhir.json
+template = fhir.base.template#current
```
This line should change to 
```
ig = fsh-generated/resources/ImplementationGuide-saraalert-fhir.json
```
once `input/ImplementationGuide-saraalert-fhir.json` is removed, as mentioned in the comment below.

tstrass replied:
https://github.com/SaraAlert/saraalert-fhir-ig/pull/6/commits/e90a101d4dc0c7a2e8219aacd6e9bf7cf6da4d62

ngfreiter reviewed (commented):
Looks good, just had one minor set of comments.

tstrass committed:
Remove redundant sushi-generated file

No longer necessary since sushi generates this file and puts it
elsewhere. Also, update the ini file accordingly to point to the
new file.

jawalonoski reviewed (commented):
This looks fine... you add the slice for the identifiers and the state local id. ✔️ 

But... where is the narrative that defines what this is and how it is used? Somewhere outside the IG?

tstrass commented:
>But... where is the narrative that defines what this is and how it is used? Somewhere outside the IG?

Good point. I guess this is the first FHIR customization we've made in the API that isn't an extension. I definitely think it should be documented here in the IG rather than somewhere else, so I'll put it somewhere here.

tstrass committed:
Improve documentation of State/Local ID

Add a short and a comment for the StateLocalId

tstrass commented:
> > But... where is the narrative that defines what this is and how it is used? Somewhere outside the IG?
> 
> Good point. I guess this is the first FHIR customization we've made in the API that isn't an extension. I definitely think it should be documented here in the IG rather than somewhere else, so I'll put it somewhere here.

Here it is @jawalonoski https://github.com/SaraAlert/saraalert-fhir-ig/pull/6/commits/aa78472a4b285072a1929ec15ea711915f19bc21

ngfreiter reviewed (approved):
This still looks good to me, not going to merge yet so that @jawalonoski has a chance to look at the updated descriptions.

jawalonoski reviewed (approved):
Looks good. Apologies for my delay in reviewing. Got lost in the holidays.
